### PR TITLE
TILA-2268 use `isReservable`-prop

### PR DESCRIPTION
**Pull request by kieferbonk**

### CODE REVIEW

File: common/src/calendar/__tests__/calendar.test.ts [hunk 1]
```diff
@@ -658,7 +784,7 @@ describe("isReservationUnitReservable", () => {
         metadataSet: {
           supportedFields: ["name"],
         },
-      } as ReservationUnitByPkType)
+      } as unknown as ReservationUnitByPkType)
```
Comment (joonatank) [inline common/src/calendar/__tests__/calendar.test.ts:787]: Not a fan of casting out type safety even in tests.

Could we just add common stuff in the file / describe level and copy it to data? ex.

```
const commonParams = {
    allowReservationsWithoutOpeningHours: false,
    authentication: ReservationUnitsReservationUnitAuthenticationChoices.Weak,
    canApplyFreeOfCharge: false,
    contactInformation: "",
    isArchived: false,
    isDraft: false,
    requireIntroduction: false,
    requireReservationHandling: false,
    reservationKind:
      ReservationUnitsReservationUnitReservationKindChoices.Direct,
    reservationStartInterval:
      ReservationUnitsReservationUnitReservationStartIntervalChoices.Interval_15Mins,
    uuid: "",
  };

isReservationUnitReservable({
        ...commonParams,
        id: "1234",
        minReservationDuration: 3600,
        maxReservationDuration: 3600,
        reservationBegins: String(addMinutes(new Date(), -10)),
        openingHours,
        metadataSet: {
          id: "",
          name: "",
          supportedFields: ["name"],
        },
})
```

That goes through typecheck without any casts at least in VS code TS 5

**Review by joonatank — CHANGES_REQUESTED** [review]

File: common/src/calendar/__tests__/calendar.test.ts [hunk 2]
```diff
@@ -93,20 +93,41 @@ describe("areSlotsReservable", () => {
...
+    },
+  ];
+
+  const openingTimes2 = [
     {
       date: format(addDays(new Date(), 7), "yyyy-MM-dd"),
       endTime: `21:00:00+${tzOffsetHoursStr}:00`,
       periods: null,
       startTime: `09:00:00+${tzOffsetHoursStr}:00`,
       state: "open",
```
[the remark this answers is not in the document]
Comment (kieferbonk) [reply common/src/calendar/__tests__/calendar.test.ts:123]: It's still in the api. I'd leave it here for posterity's sake

File: common/src/calendar/__tests__/calendar.test.ts [hunk 3]
```diff
@@ -93,20 +93,41 @@ describe("areSlotsReservable", () => {
   const tzOffset = new Date().getTimezoneOffset() / 60;
   const tzOffsetHoursStr = Math.abs(tzOffset).toString().padStart(2, "0");
 
-  const openingTimes = [
+  const openingTimes1 = [
```
[the remark this answers is not in the document]
Comment (kieferbonk) [reply common/src/calendar/__tests__/calendar.test.ts]: Good call, corrected.

File: common/src/calendar/__tests__/calendar.test.ts (hunk 1 above)
Comment (kieferbonk) [reply common/src/calendar/__tests__/calendar.test.ts]: Good call, corrected.

**Review by joonatank — APPROVED** [review]

**Pull request merged**